src/SparqlSerializer.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\rdf_entity;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\rdf_entity\Database\Driver\sparql\ConnectionInterface;
+use EasyRdf\Graph;
+
+/**
+ * Service to serialise RDF entities into various formats.
+ */
+class SparqlSerializer implements SparqlSerializerInterface {
+
+  /**
+   * The Sparql connection object.
+   *
+   * @var \Drupal\rdf_entity\Database\Driver\sparql\ConnectionInterface
+   */
+  protected $sparqlEndpoint;
+
+  /**
+   * The SPARQL graph handler service.
+   *
+   * @var \Drupal\rdf_entity\RdfGraphHandlerInterface
+   */
+  protected $graphHandler;
+
+  /**
+   * Instantiates a new RdfSerializer object.
+   *
+   * @param \Drupal\rdf_entity\Database\Driver\sparql\ConnectionInterface $sparqlEndpoint
+   *   The SPARQL connection object.
+   * @param \Drupal\rdf_entity\RdfGraphHandlerInterface $graph_handler
+   *   The SPARQL graph handler service.
+   */
+  public function __construct(ConnectionInterface $sparqlEndpoint, RdfGraphHandlerInterface $graph_handler) {
+    $this->sparqlEndpoint = $sparqlEndpoint;
+    $this->graphHandler = $graph_handler;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function serializeEntity(ContentEntityInterface $entity, string $format = 'turtle'): string {
+    $graph_uri = $this->graphHandler->getBundleGraphUri($entity->getEntityTypeId(), $entity->bundle(), $entity->graph->target_id);
+    $entity_id = $entity->id();
+
+    $query = <<<Query
+SELECT ?s ?p ?o 
+WHERE {
+  {
+    GRAPH <{$graph_uri}> {
+      ?s ?p ?o .
+      VALUES ?s { <{$entity_id}> } .
+    }
+  }
+  UNION {
```

Comment:
I know that this ticket is simply moving objects around but we might need to see on this query a bit more. This mainly second clause after UNION, mainly exports all triples that have this object as a reference. Meaning that we kinda put irrelevant information on the export based on our structure, while we should probably be exporting the object itself only.

Reply:
See the ticket comment.